app/code/Magento/CatalogSearch/Model/Adapter/Mysql/BaseSelectStrategy/BaseSelectAttributesSearchStrategy.php

```diff
@@ -68,13 +67,8 @@ public function createBaseSelect(SelectContainer $selectContainer)
 
         $select->distinct()
             ->from(
-                [$mainTableAlias => $this->resource->getTableName('catalog_product_index_eav')],
+                [$mainTableAlias => $this->resource->getTableName('catalog_product_entity')],
```

Comment:
I'm not sure, changing the base table is a good idea in this case.
It might lead to the additional issues with layered navigation or search by other non-static product attributes that are not included in `catalog_product_entity` table.